pipeline/delivlib.ts

```diff
@@ -81,11 +78,66 @@ export class DelivLibPipelineStack extends Stack {
   }
 }
 
+export class EcrMirrorStack extends Stack {
+  public readonly superchainRepo: ecr.IRepository;
+
+  constructor(scope: App, id: string, props?: StackProps) {
+    super(scope, id, props);
+
+    const superchainSource = delivlib.MirrorSource.fromDockerHub(DOCKERHUB_SUPERCHAIN);
+
+    const ecrMirror = new delivlib.EcrMirror(this, 'Default', {
+      dockerHubCreds: {
+        secret: secret.Secret.fromSecretArn(this, 'DockerHubCreds', 'arn:aws:secretsmanager:us-east-1:712950704752:secret:dockerhub/ReadOnly-VXZo5Z'),
+        usernameKey: 'username',
+        passwordKey: 'password',
+      },
+      images: [
+        superchainSource,
+      ],
+    });
+
+    const repo = ecrMirror.ecrRepository(superchainSource);
+    if (!repo) {
+      throw new Error('Cannot find ECR mirror repository for "jsii/superchain"');
+    }
+    this.superchainRepo = repo;
+  }
+}
+
+export class EcrMirrorAspect implements IAspect {
+  constructor(private readonly superchainRepo: ecr.IRepository) {}
+
+  public visit(construct: IConstruct) {
+    if (construct instanceof codebuild.Project) {
+      const cfnproject = construct.node.defaultChild as codebuild.CfnProject;
+      if (!Token.isUnresolved(cfnproject.environment)) {
+        const env = cfnproject.environment as codebuild.CfnProject.EnvironmentProperty;
+        if (env.image === 'jsii/superchain' || env.image === 'jsii/superchain:latest') {
+          cfnproject.environment = {
+            ...env,
+            image: codebuild.LinuxBuildImage.fromEcrRepository(this.superchainRepo).imageId,
+          };
+          this.superchainRepo.grantPull(construct);
+          construct.grantPrincipal.addToPrincipalPolicy(new iam.PolicyStatement({
+            actions: ['ecr:GetAuthorizationToken'],
```

Comment:
This knowledge feels more appropriate in the mirror itself. Perhaps the mirror should offer `mirror.grantPullRepository` ? or `mirror.grantPullImage`.

Reply:
Why should the `grantPull()` go into the mirror construct? It's correctly placed as an API of ECR `Repository` construct.

Reply:
To hide the IAM knowledge from the caller, but I agree aws/aws-cdk#11783 is better.